src/types/Course/Section.ts

```diff
@@ -32,11 +33,75 @@ export const CourseQuery = extendType({
   type: 'Query',
   definition(t) {
     t.list.nonNull.field('courses', {
+      // type is a list of coursesections
       type: CourseSection,
-      description: 'Get a list of courses for a given term',
+      description: 'Get a list of courses for a given term and/or year',
       args: {
         term: arg({ type: Term }),
+        year: arg({ type: 'Int' }),
       },
+      resolve: async (_, args, ctx) => {
+        const { term, year } = args;
+        const { prisma } = ctx;
+        
+        let courses: Course[] = [];
+        let courseSections: any  = [];
+
+        // if term and year are provided, get courses for that term
+        if (term && year){
+          courses = await (prisma as PrismaClient).course.findMany({
```

Comment:
Instead of declaring the variable `courses` above, setting it to the appropriate values, then returning at the end, you can just return at each step of the way.
I.e.
```
if term && year
   return <insert prisma query here>

else if term
  return <insert prisma query here>
```
and so on

Reply:
also I'll add if you're returning each way you don't need `else` statements, just a bunch of ifs

Reply:
Good catch